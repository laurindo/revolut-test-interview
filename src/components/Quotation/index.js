import React from 'react';
import PropTypes from 'prop-types';
import './quotation.css'

const Quotation = ({ symbol, currency, value }) => (
  <small className="quotation-text"> { symbol } { value } { currency }</small>
);

Quotation.propTypes = {
  symbol: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
};

export default Quotation;
