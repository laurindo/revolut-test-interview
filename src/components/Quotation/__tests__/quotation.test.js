import React from 'react';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Quotation from '../index';

configure({ adapter: new Adapter() });

describe('Quotation Component', () => {
  test('should render the component', () => {
    const component = shallow(<Quotation symbol="$" currency="USD" value={"12.99"} />);
    expect(component).toMatchSnapshot();
    expect(component.find('small')).toBeDefined();
    expect(component.find('small').get(0).props.className).toEqual('quotation-text');
    expect(component.find('small').text()).toEqual(' $ 12.99 USD');
  });
});
