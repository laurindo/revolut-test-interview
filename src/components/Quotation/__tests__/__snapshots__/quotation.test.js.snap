exports[`Quotation Component should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Quotation
    currency="USD"
    symbol="$"
    value="12.99" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        "$",
        " ",
        "12.99",
        " ",
        "USD",
      ],
      "className": "quotation-text",
    },
    "ref": null,
    "rendered": Array [
      " ",
      "$",
      " ",
      "12.99",
      " ",
      "USD",
    ],
    "type": "small",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "$",
          " ",
          "12.99",
          " ",
          "USD",
        ],
        "className": "quotation-text",
      },
      "ref": null,
      "rendered": Array [
        " ",
        "$",
        " ",
        "12.99",
        " ",
        "USD",
      ],
      "type": "small",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
